tools/submission/submission-checker.py

```diff
@@ -732,7 +732,15 @@ def check_power_dir(power_path, ranging_path, testing_path, more_power_check=Fal
         log.error("%s has no power samples falling in power range: %s - %s", spl_fname, power_begin, power_end)
         is_valid = False
     else:
-        power = sum(power_list) / len(power_list)
+        avg_power = sum(power_list) / len(power_list)
+        if scenario_fixed in ["Offline", "Server"]:
+            # In Offline and Server scenarios, the power metric is in W.
+            power = avg_power
+        elif scenario_fixed in ["SingleStream", "MultiStream"]:
+            # In SingleStream and MultiStream scenarios, the power metric is in J/sample.
+            power_duration = (power_end - power_begin).total_seconds()
+            num_samples = mlperf_log["generated_query_count"] * mlperf_log["generated_samples_per_query"]
```

Comment:
Hold on. Are we sure that these two are reliable? Does SingleStream finishes all the generated queries or does it end early when min_duration is reached? 

@psyhtest @guschmue Do you know?

Reply:
I ended up using `mlperf_log["result_qps_with_loadgen_overhead"] * power_duration`